src/seedSale/seedSale.ts

```diff
@@ -134,8 +137,23 @@ export class SeedSale {
   @computedFrom("seed.userFundingTokenBalance", "fundingTokenToPay")
   get userCanPay(): boolean { return this.seed.userFundingTokenBalance?.gt(this.fundingTokenToPay ?? "0"); }
 
-  @computedFrom("maxFundable", "seed.userFundingTokenBalance")
-  get maxUserCanPay(): BigNumber { return this.maxFundable.lt(this.seed.userFundingTokenBalance || "0") ? this.maxFundable : this.seed.userFundingTokenBalance; }
+  @computedFrom("targetClass.individualCap", "targetClass.classCap", "maxFundable", "seed.userFundingTokenBalance")
+  get maxUserCanPay(): BigNumber {
+    // /* prettier-ignore */ console.log(">>>> _ >>>> ~ file: seedSale.ts ~ line 143 ~ this.targetClass.individualCap", this.targetClass.individualCap);
+    // /* prettier-ignore */ console.log(">>>> _ >>>> ~ file: seedSale.ts ~ line 145 ~ this.targetClass.classCap", this.targetClass.classCap);
+    // /* prettier-ignore */ console.log(">>>> _ >>>> ~ file: seedSale.ts ~ line 147 ~ this.seed.userFundingTokenBalance", this.seed.userFundingTokenBalance);
+    // /* prettier-ignore */ console.log(">>>> _ >>>> ~ file: seedSale.ts ~ line 156 ~ this.maxFundable", this.maxFundable);
+    const min = this.bigNumberService.min([
+      this.targetClass.individualCap,
+      this.targetClass.classCap,
+      this.seed.userFundingTokenBalance,
+      this.maxFundable,
+    ]);
```

Comment:
these should all have the same decimals?
cc @giladt

Reply:
Not exactly sure which values these are, but in general you are right, all should have 18 decimals.

In regards to the USDC example and the different precisions, these should indeed be normalized to 18 decimals. This should be easily possible through EthersJS ‘parseUnits’  functionality